File: Java/2138. Divide a String Into Groups of Size k.java
class Solution {
    public String[] divideString(String s, int k, char fill) {
        int len = s.length();
        List<String> ans = new ArrayList<>();
        int i = 0, cur = 0;

        while (cur < len) {
            ans.add(s.substring(cur, Math.min(cur + k, len)));
            i++;
            cur += k;
        }

        String last = ans.get(ans.size() - 1);
        if (last.length() < k) {
            last += String.valueOf(fill).repeat(k - last.length());
            ans.set(ans.size() - 1, last);
        }

        return ans.toArray(new String[0]);
    }
}

File: Python3/2138. Divide a String Into Groups of Size k.py
class Solution:
    def divideString(self, s: str, k: int, fill: str) -> List[str]:
        ans = []
        length = len(s)

        for i in range(0, length, k):
            if i + k <= length:
                ans.append(s[i: i + k])
            else:
                size = length - i
                tmp = s[i:] + fill * (k - size % k)
                ans.append(tmp)

        return ans